api/envoy/extensions/filters/http/ext_proc/v3alpha/ext_proc.proto

```diff
@@ -0,0 +1,89 @@
+// It communicates with an external gRPC service that can use it to do a variety of things
+// with the request and response:
+//
+// * Access and modify the HTTP headers on the request, response, or both
+// * Access and modify the HTTP request and response bodies
+// * Access and modify the dynamic stream metadata
+// * Immediately send an HTTP response downstream and terminate other processing
+//
+// The filter communicates with the server using a gRPC bidirectional stream. After the initial
+// request, the external server is in control over what additional data is sent to it
+// and how it should be processed.
+//
+// By implementing the protocol specified by the stream, the external server can choose:
+//
+// * Whether it receives the response message at all
+// * Whether it receives the message body at all, in separate chunks, or as a single buffer
+// * Whether subsequent HTTP requests are transmitted synchronously or whether they are
+//   sent asynchronously.
+//
+// All of this together allows a server to process the filter traffic in fairly
+// sophisticated ways. For example:
+//
+// * A server may choose to examine all or part of the HTTP message bodies depending
+//   on the content of the headers.
+// * A server may choose to immediately reject some messages based on their HTTP
+//   headers (or other dynamic metadata) and more carefully examine others
+// * A server may asynchronously monitor traffic coming through the filter by inspecting
+//   headers, bodies, or both, and then decide to switch to a synchronous processing
+//   mode, either permanently or temporarily.
+
+// [#next-free-field: 8]
+message ExternalProcessor {
+  // Configuration for the gRPC service that the filter will communicate with.
+  // The filter supports both the "Envoy" and "Google" gRPC clients.
+  config.core.v3.GrpcService grpc_service = 1;
+
+  // The version of the API supported by the gRPC service.
+  config.core.v3.ApiVersion transport_api_version = 2
+      [(validate.rules).enum = {defined_only: true}];
+
+  // If true, then do nothing if the remote service RPC fails.
+  // Otherwise, return a 500 error if the gRPC call fails.
+  bool failure_mode_allow = 3;
+
+  // For each property name, select a matching property value to attach to the
+  // request message. Property names are the same ones supported in the WebAssembly
+  // runtime.
+  repeated string request_properties = 4;
+
+  // For each property name, select a matching property value to attach to the
+  // response message.
+  repeated string response_properties = 5;
+
+  // Specifies which parts of the HTTP request and response are sent to the remote
+  // server by default, whether they are sent synchronously or asynchronously,
+  // and whether the body is sent in chunks or buffers. The default processing
+  // mode sends request and response headers synchronously and nothing else.
```

Comment:
I find the phrasing on this one confusing.  I'd read synchronous send as blocking Envoy until an RPC is received rather than blocking the stream.
I think we can just say "gives default options for how headers and body are sent, see ProcessingMode for details"

Reply:
I'm all for making the doc more readable like this.